pages/03.Howtos/15.repartition-vm-disk/docs.en.md

```diff
@@ -0,0 +1,255 @@
+---
+title: 'How to repartition VM disk'
+date: '21-10-2020 16:27'
+taxonomy:
+    category:
+        - docs
+---
+
+## How to repartition VM disk
+
+### Overview
+
+This Document will show you the essential steps to repartition your VM disk. For this purpose we are using an Ubuntu 18.04 server as example with the `m1c.tiny` [flavor](../../04.Reference/03.compute/docs.en.md).
+
+
+### Prerequisites
+
+* You need to have the login data for the SysEleven Stack API (user name and passphrase)
+* Knowledge how to utilise a terminal/SSH and SSH-keys.
+
+### Automatically configure partitions
+
+By default [cloudinit](https://cloudinit.readthedocs.io/en/latest/) will grow your VM image partition size to the complete size of the ephemeral disk (which is defined by the flavor you used for your VM). To avoid this behavior and use the spare space for a new partition we will use following snippet :
+
+```shell
+#cloud-config
+growpart:
+  mode: off
+runcmd:
+# Parted asks whether to fix /dev/vda to use all available space, we reply "Fix" and continue
+# Resize primary partition to 10GB
+- "echo Fix > /tmp.txt"
+- "echo 1 >> /tmp.txt"
+- "echo yes >> /tmp.txt"
+- "echo 10GB >> /tmp.txt"
+- "parted ---pretend-input-tty /dev/vda resizepart < /tmp.txt"
```

Comment:
small detail: you could create the multi-line file in a slightly shorter way: `(echo Fix; echo 1; echo yes; echo 10GB) > /tmp/txt`
or even `(echo Fix; echo 1; echo yes; echo 10GB) | parted ---pretend-input-tty /dev/vda resizepart` if you like.

Reply:
`echo -e "Fix\n1\nyes\n10GB" | parted…`
`printf "Fix\n1\nyes\n10GB\n" | parted…`
`printf "%s\n%s\n%s\n%s\n" "Fix" "1" "yes" "10GB" | parted…`

Reply:
I didn't want to mention the `echo -e` since that's not so portable across various systems :)

Reply:
Layout-wise, I like the last version with the here-document. Lots of other variants just glue the replies to a lot of separating stuff, so you sort of lose the important parts in the sea of characters. On the other hand, the here document may be a bit hard to put in yaml form, because it is a multi-line item.

Reply:
you can do that with

```yaml
runcmd: |
  parted ... <<-EOI
    fix
    1
    yes
  EOI

Reply:
I will go with printf then, thx for pointing the issues out =)